plugin/docker/root.go

```diff
@@ -37,9 +36,17 @@ func (r *Root) Init(map[string]interface{}) error {
 	return nil
 }
 
+// Schema returns the root's schema
+func (r *Root) Schema() *plugin.EntrySchema {
```

Comment:
I wonder if we should provide a `NewRoot` so that plugin authors don't have to remember to do this.

However, this doesn't seem too bad. I guess I can also remove the code in plugin.Registry that sets roots to singletons + their labels to their name.

Reply:
🤷‍♂ seems a lot to avoid something that's needed everywhere.

Reply:
Yeah, not worth it.